drivers/clock_control/clock_stm32_ll_u5.c

```diff
@@ -226,152 +197,76 @@ static void set_regu_voltage(uint32_t hclk_freq)
 	}
 }
 
-/*
- * Unconditionally switch the system clock source to HSI.
- */
 __unused
-static void clock_switch_to_hsi(uint32_t ahb_prescaler)
-{
-	/* Enable HSI if not enabled */
-	if (LL_RCC_HSI_IsReady() != 1) {
-		/* Enable HSI */
-		LL_RCC_HSI_Enable();
-		while (LL_RCC_HSI_IsReady() != 1) {
-		/* Wait for HSI ready */
-		}
-	}
-
-	/* Set HSI as SYSCLCK source */
-	LL_RCC_SetSysClkSource(LL_RCC_SYS_CLKSOURCE_HSI);
-	LL_RCC_SetAHBPrescaler(ahb_prescaler);
-	while (LL_RCC_GetSysClkSource() != LL_RCC_SYS_CLKSOURCE_STATUS_HSI) {
-	}
-}
-
-static void set_up_clk_msis(void)
+static int set_up_plls(void)
 {
-#if defined(STM32_SYSCLK_SRC_MSIS) || defined(STM32_PLL_SRC_MSIS)
-	/* Set MSIS Range */
-	LL_RCC_MSI_EnableRangeSelection();
+#if defined(STM32_PLL_ENABLED)
+	int r;
+	uint32_t vco_input_range;
 
-	LL_RCC_MSIS_SetRange(STM32_MSIS_RANGE << RCC_ICSCR1_MSISRANGE_Pos);
+	LL_RCC_PLL1_Disable();
 
-	if (IS_ENABLED(STM32_MSIS_PLL_MODE)) {
-		BUILD_ASSERT(STM32_LSE_ENABLED,
-			"MSI Hardware auto calibration needs LSE clock activation");
-		/* Enable MSI hardware auto calibration */
-		LL_RCC_MSI_EnablePLLMode();
+	/* Configure PLL source */
+	/* Can be HSE , HSI  MSI */
+	if (IS_ENABLED(STM32_PLL_SRC_HSE)) {
+		/* Main PLL configuration and activation */
+		LL_RCC_PLL1_SetMainSource(LL_RCC_PLL1SOURCE_HSE);
+	} else if (IS_ENABLED(STM32_PLL_SRC_MSIS)) {
+		/* Main PLL configuration and activation */
+		LL_RCC_PLL1_SetMainSource(LL_RCC_PLL1SOURCE_MSIS);
+	} else if (IS_ENABLED(STM32_PLL_SRC_HSI)) {
+		/* Main PLL configuration and activation */
+		LL_RCC_PLL1_SetMainSource(LL_RCC_PLL1SOURCE_HSI);
+	} else {
+		return -ENOTSUP;
 	}
 
-	/* Set MSIS Range */
-	LL_RCC_MSIS_Enable();
-
-	/* Wait till MSIS is ready */
-	while (LL_RCC_MSIS_IsReady() != 1) {
+	r = get_vco_input_range(STM32_PLL_M_DIVISOR, &vco_input_range);
+	if (r < 0) {
+		return r;
 	}
 
-#endif /* STM32_SYSCLK_SRC_MSIS || STM32_PLL_SRC_MSIS */
-}
+	LL_RCC_PLL1_SetDivider(STM32_PLL_M_DIVISOR);
 
-/*
- * Configure PLL as source of SYSCLK
- */
-void config_src_sysclk_pll(LL_UTILS_ClkInitTypeDef s_ClkInitStruct)
-{
-#ifdef STM32_SYSCLK_SRC_PLL
-	LL_UTILS_PLLInitTypeDef s_PLLInitStruct;
+	LL_RCC_PLL1_SetVCOInputRange(vco_input_range);
 
-	/* configure PLL input settings */
-	config_pll_init(&s_PLLInitStruct);
+	LL_RCC_PLL1_SetN(STM32_PLL_N_MULTIPLIER);
 
-	/*
-	 * Switch to HSI and disable the PLL before configuration.
-	 * (Switching to HSI makes sure we have a SYSCLK source in
-	 * case we're currently running from the PLL we're about to
-	 * turn off and reconfigure.)
-	 *
-	 * Don't use s_ClkInitStruct.AHBCLKDivider as the AHB
-	 * prescaler here. In this configuration, that's the value to
-	 * use when the SYSCLK source is the PLL, not HSI.
-	 */
-	clock_switch_to_hsi(LL_RCC_SYSCLK_DIV_1);
-	LL_RCC_PLL1_Disable();
+	/* FRACN disable DIVP,DIVQ,DIVR enable*/
+	LL_RCC_PLL1FRACN_Disable();
```

Comment:
changing the PLL1 config bits requires the PLL to be disabled, 
the LL_RCC_PLL1FRACN_Disable() does not guarantee that  _PLL1ON = 0 and PLL1RDY = 0_
This is done by  the `LL_RCC_PLL1_Disable`();   and evenmore with a check on `LL_RCC_PLL1_IsReady`

Reply:
`LL_RCC_PLL1_Disable()` is called at the beginning of this function.
`LL_RCC_PLL1FRACN_Disable` here is only meant to disable FRACN and reused from existing code (could be optionally cleaned up).
But I agree that comment is misleading, will fix.